qcodes/instrument_drivers/tektronix/AWGFileParser.py

```diff
@@ -0,0 +1,560 @@
+        m2[ii] = int(bitstring[0])
+        m1[ii] = int(bitstring[1])
+        wf[ii] = (int(bitstring[2:], base=2)-2**13)/2**13
+        # print(bitstring, int(bitstring[2:], base=2))
+
+    return wf, m1, m2
+
+
+def _unwrap(bites, fmt):
+    """
+    Helper function for interpreting the bytes from the awg file.
+
+    Args:
+        bites (bytes): a bytes object
+        fmt (str): the format string (either 's', 'h' or 'd')
+
+    Returns:
+        Union[str, int, tuple]
+    """
+
+    if fmt == 's':
+        value = bites[:-1].decode('ascii')
+    elif fmt is 'ignore':
+        value = 'Not read'
+    else:
+        value = struct.unpack('<'+fmt, bites)
+        if len(value) == 1:
+            value = value[0]
+
+    return value
+
+
+def _getendingnumber(string):
+    """
+    Helper function to extract the last number of a string
+
+    Args:
+        string (str): A .awg field name, like SEQUENCE_JUMP_23
+
+    Returns
+        (int, str): The number and the shortened string,
+          e.g. 'SEQUENCE_JUMP_23' -> (23, 'SEQUENCE_JUMP_')
+    """
+
+    num = ''
+
+    for char in string[::-1]:
+        if char.isdigit():
+            num += char
+        else:
+            break
+
+    return (int(num[::-1]), string[:-len(num)])
+
+
+awgfilepath = ('/Users/william/AuxiliaryQCoDeS/AWGhelpers/awgfiles/' +
+               'customawgfile.awg')
+
+awgfilepath2 = ('/Users/william/AuxiliaryQCoDeS/AWGhelpers/awgfiles/' +
+                'machinemadefortest.awg')
+
+
+def _parser1(awgfilepath):
+    """
+    Helper function doing the heavy lifting of reading and understanding the
+    binary .awg file format.
+
+    Args:
+        awgfilepath (str): The absolute path of the awg file to read
+
+    Returns:
+        (dict, list, list): Instrument settings, waveforms, sequencer settings
+    """
+
+    instdict = {}
+    waveformlist = [[], []]
+    sequencelist = [[], []]
+
+    with open(awgfilepath, 'rb') as fid:
+
+        while True:
+            chunk = fid.read(8)
+            if not chunk:
+                break
+
+            (namelen, valuelen) = struct.unpack('<II', chunk)
+
+            rawname = fid.read(namelen)
+            rawvalue = fid.read(valuelen)
+
+            name = rawname[:-1].decode('ascii')  # remove NULL termination char
+
+            if name.startswith('WAVEFORM'):
+
+                namestop = name[name.find('_')+1:].find('_')+name.find('_')
+                lookupname = name[:namestop+1]
+
+                if 'DATA' in name:
+                    fmtstr = '{}H'.format(wfmlen)
+                    AWG_FILE_FORMAT_WAV['WAVEFORM_DATA'] = fmtstr
+
+                value = _unwrap(rawvalue, AWG_FILE_FORMAT_WAV[lookupname])
+                (number, barename) = _getendingnumber(name)
+                fieldname = barename + '{}'.format(number-20)
+                waveformlist[0].append(fieldname)
+                waveformlist[1].append(value)
+
+                if 'LENGTH' in name:
```

Comment:
Are we sure that 'LENGTH' always comes before 'DATA' such that `wfmlen` is always defined above